compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryRegional.java

```diff
@@ -0,0 +1,107 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateDiskSecondaryRegional {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // The project that contains the primary disk.
+    String primaryProjectId = "PRIMARY_PROJECT_ID";
+    // The project that contains the secondary disk.
+    String secondaryProjectId = "SECONDARY_PROJECT_ID";
+    // Name of the primary disk you want to use.
+    String primaryDiskName = "PRIMARY_DISK_NAME";
+    // Name of the disk you want to create.
+    String secondaryDiskName = "SECONDARY_DISK_NAME";
+    // Name of the region in which your primary disk is located.
+    // Learn more about zones and regions:
+    // https://cloud.google.com/compute/docs/disks/async-pd/about#supported_region_pairs
+    String primaryDiskRegion = "us-central1";
+    // Name of the region in which you want to create the secondary disk.
+    String secondaryDiskRegion = "us-east1";
+    // Size of the new disk in gigabytes.
+    // Learn more about disk requirements:
+    // https://cloud.google.com/compute/docs/disks/async-pd/configure?authuser=0#disk_requirements
+    long diskSizeGb = 30L;
+    // The type of the disk you want to create. This value uses the following format:
+    // "projects/{projectId}/zones/{zone}/diskTypes/
+    // (pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+    String diskType = String.format(
+        "projects/%s/regions/%s/diskTypes/pd-balanced", secondaryProjectId, secondaryDiskRegion);
+
+    createDiskSecondaryRegional(primaryProjectId, secondaryProjectId, primaryDiskName,
+        secondaryDiskName, primaryDiskRegion, secondaryDiskRegion, diskSizeGb, diskType);
+  }
+
+  // Creates a secondary disk in a specified region.
+  public static Status createDiskSecondaryRegional(String projectId,
+       String secondaryProjectId, String primaryDiskName, String secondaryDiskName,
+       String primaryDiskRegion, String secondaryDiskRegion, long diskSizeGb, String diskType)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // An iterable collection of zone names in which you want to keep
+    // the new disks' replicas. One of the replica zones of the clone must match
+    // the zone of the source disk.
+    List<String> replicaZones = Arrays.asList(
+        String.format("projects/%s/zones/%s-c", secondaryProjectId, secondaryDiskRegion),
+        String.format("projects/%s/zones/%s-b", secondaryProjectId, secondaryDiskRegion));
+
+    String primaryDiskSource = String.format("projects/%s/regions/%s/disks/%s",
+        projectId, primaryDiskRegion, primaryDiskName);
+
+    DiskAsyncReplication asyncReplication = DiskAsyncReplication.newBuilder()
+        .setDisk(primaryDiskSource)
+        .build();
```

Comment:
Kindly move this code within the try block

Reply:
Thank you! Fixed